src/amqp/publisher.ts

```diff
@@ -1,25 +1,34 @@
 import amqp from 'amqplib';
 import Debug from 'debug';
 
-import { PubSubAMQPConfig, Exchange, Queue } from './interfaces';
+import { PubSubAMQPConfig, Exchange } from './interfaces';
 
 export class AMQPPublisher {
   private connection: amqp.Connection;
-  private exchange: Exchange;
+  private exchange: Exchange = {
+    options: {
+      durable: false,
+      autoDelete: false
+    }
+  };
   private channel: amqp.Channel | null = null;
 
   constructor(
-    private config: PubSubAMQPConfig,
+    config: PubSubAMQPConfig,
     private logger: Debug.IDebugger
   ) {
     this.connection = config.connection;
-    this.exchange = config.exchange;
+    this.exchange = { ...config.exchange };
   }
 
   public async publish(routingKey: string, data: any): Promise<void> {
     const channel = await this.getOrCreateChannel();
-    await channel.assertExchange(this.exchange.name, this.exchange.type, { ...this.exchange.options });
-    await channel.publish(this.exchange.name, routingKey, Buffer.from(JSON.stringify(data)));
+    await channel.assertExchange(
+      this.exchange.name || 'graphql_subscriptions',
```

Comment:
wondering if it would make more sense to just add the default values to the constructor instead of in all of the function calls

Reply:
Move the settings in the constructor while keeping the typings correct